### PR TITLE
[node] Fix typings for various child_process methods

### PR DESCRIPTION
**Pull request by thw0rted**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [npm child_process docs](https://nodejs.org/api/child_process.html)
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Required minimal changes outside of Node.  Build appears to be broken but [it's TS@next's fault](https://github.com/Microsoft/types-publisher/issues/478) as far as I can tell.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -2190,11 +2190,13 @@ declare module "child_process" {
         keepOpen?: boolean;
     }
 
+    export type StdioOptions = "pipe" | "ignore" | "inherit" | Array<("pipe" | "ipc" | "ignore" | stream.Stream | number)>;
```
Comment (Flarna) [inline types/node/index.d.ts]: `null` and `undefined` are also allowed as types in the array

Comment (thw0rted) [reply types/node/index.d.ts]: Good point, I overlooked those in the docs.

File: types/node/index.d.ts [hunk 2]
```diff
@@ -2329,32 +2331,31 @@ declare module "child_process" {
...
+        stdio?: StdioOptions;
+        windowsVerbatimArguments?: boolean;
         uid?: number;
         gid?: number;
-        windowsVerbatimArguments?: boolean;
     }
     export function fork(modulePath: string, args?: ReadonlyArray<string>, options?: ForkOptions): ChildProcess;
 
     export interface SpawnSyncOptions {
-        argv0?: string;
```
Comment (Flarna) [inline types/node/index.d.ts]: I'm quite sure that `argv0` argument is also supported for sync even if it is missing in node docs.

Comment (thw0rted) [reply types/node/index.d.ts]: It does seem pretty unlikely that `spawnsync` wouldn't end up reusing most of the logic from `spawn` (or vice versa).  I'll put the option back here and see about getting the Node docs fixed -- if it turns out you really can't, I'll open a separate PR to take it out, I guess.

Comment (thw0rted) [reply types/node/index.d.ts]: That didn't take long: [found it](https://github.com/nodejs/node/blob/master/lib/child_process.js#L490).  Definitely uses the same logic for both functions.

Comment (thw0rted) [reply types/node/index.d.ts]: [Node PR](https://github.com/nodejs/node/pull/22231) is up.

**Review by Flarna — APPROVED** [review]

**Pull request merged**